### PR TITLE
Replace catch_panic by catch_unwind

### LINKED ISSUE #212: catch_panic => catch_unwind
The Unwinding chapter refers to a `catch_panic`. As far as I could tell, that function has been replaced in 1.9 by `std::panic::catch_unwind`. Besides the name change, should this affect the content of the chapter?

### PR DESCRIPTION
**Pull request by dbdr**
Fixes #212

### CODE REVIEW

**Comment by dbdr** [comment]
It looks like the test fail because the link *checker* is broken:
```
error[E0599]: no function or associated item named `empty` found for struct `std::ptr::Unique<_>` in the current scope
```
The link itself works for me.

File: src/unwinding.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ panics can only be caught by the parent thread. This means catching a panic
 requires spinning up an entire OS thread! This unfortunately stands in conflict
 to Rust's philosophy of zero-cost abstractions.
 
-There is an API called `catch_panic` that enables catching a panic
+There is an API called [`catch_unwind`][catch_unwind] that enables catching a panic
```
Comment (lzutao) [inline src/unwinding.md:34]:
```suggestion
There is an API called [`catch_unwind`] that enables catching a panic
```

File: src/unwinding.md [hunk 2]
```diff
@@ -47,3 +47,5 @@ You must *absolutely* catch any panics at the FFI boundary! What you do at that
 point is up to you, but *something* must be done. If you fail to do this,
 at best, your application will crash and burn. At worst, your application *won't*
 crash and burn, and will proceed with completely clobbered state.
+
+[catch_unwind]: https://doc.rust-lang.org/std/panic/fn.catch_unwind.html
```
Comment (lzutao) [inline src/unwinding.md:51]:
```suggestion
[`catch_unwind`]: https://doc.rust-lang.org/std/panic/fn.catch_unwind.html
```

**Comment by Gankra** [comment]
It's now Unique::dangling: https://github.com/rust-lang/rust/commit/97a8870022ec819c8b92b4c192242726c77e19f3

but yea, was broken before you got here

**Pull request merged**